doc/reference/routers.md

```diff
@@ -0,0 +1,215 @@
+    - path: /users/{username}/hovercard
+      backend: parameter-path-backend
+    - path: /users/{username:[a-z]+}/hovercard
+      backend: regexp-backend
+    - path: /users/*
+      backend: prefix-backend
+    - path: /users/aniaan/hovercard
+      backend: full-match-backend
+```
+
+| path | Match backend |
+|------|--------------|
+|/users/aniaan/hovercard | `full-match-backend` |
+|/users/12345/hovercard | `parameter-path-backend` |
+|/users/localvar/hovercard | `regexp-backend` |
+|/users/test | `prefix-backend` |
+|/blog/bar | not match |
+
+Let's take a look at another example to deepen our understanding
+
+```yaml
+kind: HTTPServer
+name: server-demo
+port: 10080
+keepAlive: true
+https: false
+routerKind: RadixTree
+rules:
+  - paths:
+    - path: /user/{username}/123
+      backend: h1
+    - path: /user/wang/{num}
+      backend: h2
+    - path: /{type}/wang/123
+      backend: h3
+```
+
+`/user/wang/123` will match `/user/wang/{num}`, which is determined by the matching order of RadixTree, you can split the path into individual characters, each character follows the matching order of `Full match` -> `Regexp match` -> `Parameter match` -> `Prefix match`, so the final match result is `/user/wang/{num}`.
+
+RadixTree, compared with Ordered, is lost its definition order.
+
+See one more.
+
+```yaml
+kind: HTTPServer
+name: server-demo
+port: 10080
+keepAlive: true
+https: false
+routerKind: RadixTree
+- paths:
+    - path: /users/{u:[a-z]+}sername/hovercard
+      backend: h1
+    - path: /users/u{sername}/hovercard
+      backend: h2
+```
+
+Similarly, `/users/username/hovercard` will match `/users/u{sername}/hovercard` for the same reason as above, so you can try to understand it.
+
+### Rewrite
+
+Let's take a look at how RadixTree's Path Rewrite feature works. The syntax of Rewrite remains basically the same as that of Path, with a few minor differences.
+
+```yaml
+kind: HTTPServer
+name: server-demo
+port: 10080
+keepAlive: true
+https: false
+routerKind: RadixTree
+rules:
+  - paths:
+    - path: /users/{username}/hovercard
+      rewriteTarget: /api/users/{username}/card
+      backend: parameter-path-backend
+    - path: /users/{username:[a-z]+}/hovercard
+      rewriteTarget: /api/users/{username}/card
+      backend: regexp-backend
+    - path: /users/*
+      rewriteTarget: /api/users/{eg_wildcard}
+      backend: prefix-backend
+    - path: /users/aniaan/hovercard
+      rewriteTarget: /api/users/aniaan/card
+      backend: full-match-backend
+```
+
+| path | Match backend | Rewrite target|
+|------|--------------|----------------|
+|/users/aniaan/hovercard | `full-match-backend` | /api/users/aniaan/card |
+|/users/12345/hovercard | `parameter-path-backend` | /api/users/12345/card |
+|/users/localvar/hovercard | `regexp-backend` |  /api/users/localvar/card |
+|/users/test | `prefix-backend` | /api/users/test |
+
+You can see that only the wildcard `*` needs to be replaced by the built-in variable `eg_wildcard`.
```

Comment:
To be distinguished for built-in variables, how about using all capital letters as `EG_WILDCARD`

Reply:
updated